catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/DataListFragment.java

```diff
@@ -350,6 +361,209 @@ public void onItemLongClick(final UserData item, CheckableVH holder) {
+		}
+
+		for (Scene scene : sceneList) {
+			for (Sprite sprite : scene.getSpriteList()) {
+				if (sprite == currentSprite || currentScene == null) {
+					for (Script script : sprite.getScriptList()) {
+						List<Brick> flatList = new ArrayList();
+						script.addToFlatList(flatList);
+						for (Brick brick : flatList) {
+							if (brick instanceof FormulaBrick) {
+								FormulaBrick formulaBrick = (FormulaBrick) brick;
+								for (Formula formula : formulaBrick.getFormulas()) {
+									FormulaElement formulaTree = formula.getFormulaTree();
+									List<String> lists = new ArrayList<>();
+									List<String> variables = new ArrayList<>();
+									formulaTree.getVariableAndListNames(variables, lists);
+
+									if (item instanceof UserVariable && variables.contains(item.getName())) {
+										showInUseAlert(item);
+										return true;
+									} else if (item instanceof UserList && lists.contains(item.getName())) {
+										showInUseAlert(item);
+										return true;
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+		return false;
+	}
+
+	private boolean checkIfUserDataItemInDropDown(UserData item, @Nullable Scene currentScene) {
+		List<Scene> sceneList;
+		Project project = projectManager.getCurrentProject();
+		Sprite currentSprite = projectManager.getCurrentSprite();
+
+		if (currentScene == null) {
+			sceneList = project.getSceneList();
+		} else {
+			sceneList = new ArrayList<>();
+			sceneList.add(currentScene);
+		}
+
+		for (Scene scene : sceneList) {
+			for (Sprite sprite : scene.getSpriteList()) {
+				if (sprite == currentSprite || currentScene == null) {
+					for (Script script : sprite.getScriptList()) {
+						List<Brick> flatList = new ArrayList();
+						script.addToFlatList(flatList);
+						for (Brick brick : flatList) {
+							if (brick instanceof UserVariableBrickWithFormula && item instanceof UserVariable) {
+								if (((UserVariableBrickWithFormula) brick).getUserVariable() != null
+										&& item.getName() == ((UserVariableBrickWithFormula) brick).getUserVariable().getName()) {
+									if (currentScene == null) {
+										showInDropDownDialog(item, sceneList, null);
+									} else {
+										showInDropDownDialog(item, sceneList, currentSprite);
+									}
+									return true;
+								}
+							} else if (brick instanceof UserListBrick && item instanceof UserList) {
+								if (((UserListBrick) brick).getUserList() != null
+										&& item.getName() == ((UserListBrick) brick).getUserList().getName()) {
+									if (currentScene == null) {
+										showInDropDownDialog(item, sceneList, null);
+									} else {
+										showInDropDownDialog(item, sceneList, currentSprite);
+									}
+									return true;
+								}
+							} else if (brick instanceof WriteVariableOnDeviceBrick) {
+								cleanUpVariableOnDevice(project);
+							} else if (brick instanceof WriteListOnDeviceBrick) {
+								cleanUpListOnDevice(project);
```

Comment:
@gPathpp : This doesn't needs to be done, because theres already an logic implemented to distinguish between Lists and Variables.